bundles/framework/announcements/view/flyout/FlyoutCollapse.jsx

```diff
@@ -13,23 +13,26 @@ export const FlyoutCollapse = ({
             <Message messageKey={'flyout.noAnnouncements'}/>
         );
     }
+    const items = announcements.map((announcement) => {
+        const { locale, id } = announcement;
+        const { title } = Oskari.getLocalized(locale);
+        const dateRange = getDateRange(announcement);
+
+        return {
+            key: announcement.id,
+            label: title,
+            extra: <CollapseTools toolController={toolController} announcementId={id}/>,
+            children: <>
+                <AnnouncementsContent announcement={announcement}/>
+                <Divider />
+                <b><Message messageKey={'valid'} /></b>
+                <p>{dateRange}</p>
+            </>
+        };
+    });
+
     return (
-        <Collapse accordion>
-            { announcements.map((announcement) => {
-                const { locale, id } = announcement;
-                const { title } = Oskari.getLocalized(locale);
-                const dateRange = getDateRange(announcement);
-                return (
-                    <CollapsePanel header={title} key={announcement.id}
-                        extra={<CollapseTools toolController={toolController} announcementId={id}/>}>
-                        <AnnouncementsContent announcement={announcement}/>
-                        <Divider />
-                        <b><Message messageKey={'valid'} /></b>
-                        <p>{dateRange}</p>
-                    </CollapsePanel>
-                );
-            })}
-        </Collapse>
+        <Collapse accordion items={items}/>
```

Comment:
Do these have anything like renderItem-function available? We could try to change these to utilize that instead if one is available. Probably not any issue here, but some parts might benefit if we do the data-> jsx on renderItem "lazily" instead of eagerly generating the jsx for the items array.

Reply:
Specifically the permissions tab on admin-layereditor can be very heavy to render

Reply:
Fortunately that one doesn't have collapses 😛 

Lazy rendering seems to be the default, i.e. when the collapse is closed the content isn't being rendered unless forceRender is set.